bin/ch-run.c

```diff
@@ -184,6 +187,36 @@ void env_delta_append(struct env_delta **ds, enum env_action act, char *arg)
    (*ds)[i].arg = arg;
 }
 
+/* Parse and set input from --set-env */
```

Comment:
This function I think needs more commenting explaining the algorithm and its uses.